src/problem_description.cpp

```diff
@@ -8,156 +8,24 @@
-           ((spatial_dims == 2) && (in_layout == "NCHWc") && (out_layout == "NCHWc") &&
-            (weights_layout == "CHWNc"));
-}
+bool ProblemDescription::IsLayoutNCHWc() const { return conv_problem.IsLayoutNCHWc(); }
 
 void ProblemDescription::Serialize(std::ostream& stream) const
 {
-    if(!direction.IsKnown())
-        MIOPEN_THROW("!direction.IsKnown()");
-    const auto sep = '-';
-    // Problem description with default NCHW-NCHW-NCHW layout
-    // 576-4-4-1x1-192-4-4-8-1x1-2x2-3x3-0-NCHW-FP32-F
-    // Problem description with non-default layout
-    // 576-4-4-1x1-192-4-4-8-1x1-2x2-3x3-0-NHWC-NCHW-NCHW-FP32-F
-    // clang-format off
-    stream << n_inputs;
-    stream << sep << PrintDHW(sep, spatial_dims, in_depth, in_height, in_width);
-    stream << sep << PrintDHW('x', spatial_dims, kernel_size_d, kernel_size_h, kernel_size_w);
-    stream << sep << n_outputs;
-    stream << sep << PrintDHW(sep, spatial_dims, out_depth, out_height, out_width);
-    stream << sep << batch_sz;
-    stream << sep << PrintDHW('x', spatial_dims, pad_d, pad_h, pad_w);
-    stream << sep << PrintDHW('x', spatial_dims, kernel_stride_d, kernel_stride_h, kernel_stride_w);
-    stream << sep << PrintDHW('x', spatial_dims, kernel_dilation_d, kernel_dilation_h, kernel_dilation_w);
-    stream << sep << bias;
-    if ((in_layout == "NCHW" && weights_layout == "NCHW" && out_layout == "NCHW")
-        || (in_layout == "NCDHW" && weights_layout == "NCDHW" && out_layout == "NCDHW"))
-    {
-        stream << sep << in_layout;
-    } else {
-        stream << sep << in_layout;
-        stream << sep << weights_layout;
-        stream << sep << out_layout;
-    }
-    stream << sep << EncodeDataTypesForKey(in_data_type, weights_data_type, out_data_type);
-    stream << sep << (direction.IsForward() ? "F" : direction.IsBackwardData() ? "B" : "W");
-    // clang-format on
-    // New performance config entries shall come into variable/optional part of db key.
-    // This is to support backward compatibility with previous versions of databases.
-    std::ostringstream optional;
-    {
-        // Group count > 1 identifies Group/Depthwise modes.
-        if(group_counts != 1)
-            optional << 'g' << group_counts;
-    }
-    if(!optional.str().empty())
-    {
-        stream << '_' << optional.str();
-    }
-}
-
-ProblemDescription::ProblemDescription(const TensorDescriptor& in,
-                                       const TensorDescriptor& weights,
-                                       const TensorDescriptor& out,
-                                       const ConvolutionDescriptor& conv,
-                                       conv::Direction dir,
-                                       int bias_)
-    : ProblemDescription(dir == conv::Direction::Forward
-                             ? conv::ProblemDescription{in, weights, out, conv, dir, bias_}
-                             : conv::ProblemDescription{out, weights, in, conv, dir, bias_})
```

Comment:
After this PR, we do not swap xDesc and yDesc in the ProblemDescription, i.e. GetIn() always returns xDesc.

Reply:
Oh, no, we still swap descriptors.

/cc @averinevg